modules/file-system/filestore/main.tf

```diff
@@ -104,5 +104,10 @@ resource "google_filestore_instance" "filestore_instance" {
         BASIC_HDD or BASIC_SSD tiers. Otherwise the range size must be 24.
         EOT
     }
+
+    precondition {
+      condition     = var.protocol != "NFS_V4_1" || contains(["HIGH_SCALE_SSD", "ZONAL", "REGIONAL", "ENTERPRISE"], var.filestore_tier)
```

Comment:
This should fail open rather than closed in case a new tier is created. We do not know the future, and new tiers may exist that support V4.1. Replace this condition instead with what we actually know: NFS v4.1 fails on BASIC tiers.

Rewrite this as a match against tier name beginning with "BASIC" and a similarly wored error message.